docs/migration-service.md

```diff
@@ -124,94 +260,222 @@ flowchart TD
+3. **Running Backup Service** (Soft Launch): Execute `backup-cli` scripts to backup and restore keyshares during migrations
+4. **Managing Environment Variables** (Soft Launch): Ensure `MPC_BACKUP_ENCRYPTION_KEY_HEX` is consistently set on both MPC node and backup-cli
+
+> **Hard Launch**: In hard launch, the backup service runs autonomously in a TEE and requires no manual intervention from operators beyond initial registration.
 
 
 #### Cryptography
-A pair-wise key establishment scheme can be leveraged to establish a symmetric key. A suitable implementation can be found in [libsodium](https://doc.libsodium.org/) and its rust wrapper [libsodium-rs](https://docs.rs/crate/libsodium-rs/latest). Specifically, the **key exchange protocol based on X25519** would be a good fit for the given set-up (c.f. [libsodium](https://doc.libsodium.org/key_exchange) and [libsodium-rs](https://docs.rs/libsodium-rs/latest/libsodium_rs/crypto_kx/index.html)).
+
+**Key Establishment:**
 
 The key establishment scheme requires that the node as well as the backup service have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically, for each backup generation or recovery, the node and the backup service generate ephemeral keys on the `Curve25519` and publish them on the MPC smart contract. The node and the backup service can then each run a key generation protocol using their private key and the public key of the other party. 
 
 _Note: The curious reader might ask why this protocol does not simply use the NEAR public/private key pairs associated to the MPC node and the backup service. The reason for this is is twofold:_
 - _those keys are meant for signature generation._
 - _While it is true that Curve25519 used in X25519 and edwards25519 used by the NEAR blockchain are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically convert the NEAR account keys and use them for `X25519`, it is generally advised to use one key per application. This also allows us to use ephemeral keys, as opposed to static keys for the encryption. Which is desirable._
 
+The migration service uses a defense-in-depth encryption approach with two layers:
+
+**Transport Layer (TLS 1.3):**
+
+- **Authentication**: Mutual TLS (mTLS) authenticates both parties using Ed25519 P2P keys registered in the contract
+- **Key Exchange**: TLS 1.3 handshake performs Elliptic Curve Diffie-Hellman Ephemeral (ECDHE) key exchange to derive ephemeral session keys
+- **Session Encryption**: Negotiated cipher suite
+- **Forward Secrecy**: Past sessions remain secure even if long-term Ed25519 keys are compromised (ephemeral keys discarded after session)
+- Implemented using `rustls` with TLS 1.3 only ([enforced](https://github.com/near/mpc/blob/9c34614e87280932da9d6cdc3cee8537b6443b7a/crates/tls/src/constants.rs#L11) via `&rustls::version::TLS13`)
```

Comment:
What is the reason for explaining TLS here? I find this a bit noisy. This could probably be shortened by just stating that we use TLS with mutual authentication and that we leverage the blockchain as a public key infrastructure.